### PR TITLE
ref: Make public APIs Swift friendly

### LINKED ISSUE #1844: Check public APIs for Swift friendliness
### Description

Go through all public APIs and check, if they conform to [Swift naming guidelines](https://www.swift.org/documentation/api-design-guidelines/).

- [ ] https://github.com/getsentry/sentry-cocoa/issues/1842

### PR DESCRIPTION
**Pull request by brustolin**
## :scroll: Description

- Renamed `SentrySDK.addBreadcrumb(crumb:)` to `SentrySDK.addBreadcrumb(_ crumb:)`
- Renamed `SentryClient.add(_ crumb:)` to `SentryClient.addBreadcrumb(_ crumb:)`
- Renamed `SentryClient.add(_ attachment:)` to `SentryClient.addAttachment(_ attachment:)`
- Renamed `SentryClient.apply(to:)` to `SentryClient.applyTo(session:)`
- Renamed `SentryClient.apply(to:maxBreadcrumb:)` to `SentryClient.applyTo(event:maxBreadcrumbs:)`

## :bulb: Motivation and Context

close #1844 

## :pencil: Checklist

- [x] I reviewed the submitted code
- [ ] I added tests to verify the changes
- [ ] I updated the docs if needed
- [ ] Review from the native team if needed
- [ ] No breaking changes

### CODE REVIEW

**Comment by kevinrenskers** [comment]
> * Renamed `SentryClient.apply(to:)` to `SentryClient.applyTo(session:)`
> * Renamed `SentryClient.apply(to:maxBreadcrumb:)` to `SentryClient.applyTo(event:maxBreadcrumbs:)`

Big fan of most changes, but these don't feel like good improvements to me. `SentryClient.apply(to:)` feels more "Swifty".

**Comment by brustolin** [comment]
> Big fan of most changes, but these don't feel like good improvements to me. SentryClient.apply(to:) feels more "Swifty".

I  find it odd the two apply functions have a total different kind of target. 
In this case in particular they also have different signatures, so, someone familiar with the SDK may know the target just by looking. 
But I think we should keep "clarity". Maybe `apply(toSession:)` `apply(toEvent:)`.

**Comment by kevinrenskers** [comment]
Ah sorry I missed the part where one applies to a session and the other to an event. I think your change makes sense, but I also like your latest suggestion 👍

**Review by kevinrenskers — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,15 @@
...
 This version adds a dependency on Swift.
 
+### Breaking Changes
+
+- Make public APIs Swift friendly (#2416)
+    - Renamed `SentrySDK.addBreadcrumb(crumb:)` to `SentrySDK.addBreadcrumb(_ crumb:)`
+    - Renamed `SentryClient.add(_ crumb:)` to `SentryClient.addBreadcrumb(_ crumb:)`
+    - Renamed `SentryClient.add(_ attachment:)` to `SentryClient.addAttachment(_ attachment:)`
+    - Renamed `SentryClient.apply(to:)` to `SentryClient.applyTo(session:)`
+    - Renamed `SentryClient.apply(to:maxBreadcrumb:)` to `SentryClient.applyTo(event:maxBreadcrumbs:)`
```
Comment (philipphofmann) [inline CHANGELOG.md:14]: `m`: Please use the present tense in the Changelog.
```suggestion
    - Rename `SentrySDK.addBreadcrumb(crumb:)` to `SentrySDK.addBreadcrumb(_ crumb:)`
    - Rename `SentryClient.add(_ crumb:)` to `SentryClient.addBreadcrumb(_ crumb:)`
    - Rename `SentryClient.add(_ attachment:)` to `SentryClient.addAttachment(_ attachment:)`
    - Rename `SentryClient.apply(to:)` to `SentryClient.applyTo(session:)`
    - Rename `SentryClient.apply(to:maxBreadcrumb:)` to `SentryClient.applyTo(event:maxBreadcrumbs:)`
```

Comment (philipphofmann) [inline CHANGELOG.md:14]: `m`: Scope and not Client
```suggestion
    - Renamed `SentryScope.add(_ crumb:)` to `SentryClient.addBreadcrumb(_ crumb:)`
    - Renamed `SentryScope.add(_ attachment:)` to `SentryClient.addAttachment(_ attachment:)`
    - Renamed `SentryScope.apply(to:)` to `SentryClient.applyTo(session:)`
    - Renamed `SentryScope.apply(to:maxBreadcrumb:)` to `SentryClient.applyTo(event:maxBreadcrumbs:)`
```

File: Sources/Sentry/Public/SentryScope.h [hunk 2]
```diff
@@ -103,10 +104,11 @@ NS_SWIFT_NAME(Scope)
...
  * Adds the Scope to the event
  */
-- (SentryEvent *__nullable)applyToEvent:(SentryEvent *)event
-                          maxBreadcrumb:(NSUInteger)maxBreadcrumbs;
+- (nullable SentryEvent *)applyToEvent:(SentryEvent *)event
+                         maxBreadcrumb:(NSUInteger)maxBreadcrumbs
+    NS_SWIFT_NAME(applyTo(event:maxBreadcrumbs:));
 
-- (void)applyToSession:(SentrySession *)session;
+- (void)applyToSession:(SentrySession *)session NS_SWIFT_NAME(applyTo(session:));
```
Comment (philipphofmann) [inline Sources/Sentry/Public/SentryScope.h:111]: `m`: I think those two don't need to be public. I would make them internal by moving them to SentryScope+Private.h

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,15 @@
 
 This version adds a dependency on Swift.
 
+### Breaking Changes
+
+- Make public APIs Swift friendly (#2416)
+    - Renamed `SentrySDK.addBreadcrumb(crumb:)` to `SentrySDK.addBreadcrumb(_ crumb:)`
+    - Renamed `SentryClient.add(_ crumb:)` to `SentryClient.addBreadcrumb(_ crumb:)`
+    - Renamed `SentryClient.add(_ attachment:)` to `SentryClient.addAttachment(_ attachment:)`
```
Comment (philipphofmann) [inline CHANGELOG.md:12]: `m`: Does two methods especially `add(_ crumb:)` is going to break almost every user. Can we keep the old method and mark it as deprecated, please?

**Review by philipphofmann — COMMENTED** [review]
Thanks for doing this, @brustolin.

**Pull request merged**